great_expectations/datasource/new_datasource.py

```diff
@@ -293,6 +294,12 @@ def _validate_batch_request(self, batch_request: BatchRequest):
                 """
             )
 
+        if not (batch_request.data_connector_name in self.data_connectors.keys()):
```

Comment:
Just personal preference
```suggestion
        if batch_request.data_connector_name not in self.data_connectors.keys():
```